### PR TITLE
Let the install script continue when --non-interactive is passed and no pre-existing template directory is found

### PR DESCRIPTION
**Pull request by AlexGoris-Cashforce**
As requested this has been separated from #19 .
I've added a custom template directory to the install script.

By passing `--template-dir` to the install script, the user can specify
which directory should be used for the teomplate directory

Also, when `--non-interactive` is passed (regardless of the above!) the script will continue with setting a new template directory if it can't find a pre-existing one, either to the directory specified by the user, or to a default directory.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ The supported hooks are listed below. Refer to the [Git documentation](https://g
...
+The `.ignore` files allow excluding files from being treated as a hook script. They allow _glob_ filename patterns, empty lines and comments, where the line starts with a `#` character. In the above example, one of the `.ignore` files should contain `*.md` to exclude the `pre-commit/docs.md` Markdown file. The `.githooks/.ignore` file applies to each of the hook directories, and should still define filename patterns, `*.txt` instead of `**/*.txt` for example. If there is a `.ignore` file both in the hook type folder and in `.githooks`, the files whose filename matches any pattern from either of those two files will be excluded. You can also manage `.ignore` files using the [command line helper](https://github.com/rycus86/githooks/blob/master/docs/command-line-tool.md) tool, and running `git hooks ignore <pattern>`.
```
Comment (rycus86) [inline README.md]: Why this change from `*` to `_` ?

File: README.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ To try and make things a little bit more secure, Githooks checks if any new hook
...
 
-There is a caveat worth mentioning: if a terminal *(tty)* can't be allocated, then the default action is to accept the changes or new hooks. Let me know in an issue if you strongly disagree, and you think this is a big enough risk worth having slightly worse UX instead.
+There is a caveat worth mentioning: if a terminal _(tty)_ can't be allocated, then the default action is to accept the changes or new hooks. Let me know in an issue if you strongly disagree, and you think this is a big enough risk worth having slightly worse UX instead.
```
Comment (rycus86) [inline README.md]: Same as above, was this an intentional change?

File: README.md [hunk 3]
```diff
@@ -137,16 +137,16 @@ Githooks will set up a Git alias for `git hooks <cmd>` for you, that enables you
 The commands below fetch and execute the [install.sh](install.sh) script from this repository. It will:
 
 1. Find out where the Git templates directory is
-    1. From the `$GIT_TEMPLATE_DIR` environment variable
-    2. With the `git config --get init.templateDir` command
-    3. Checking the default `/usr/share/git-core/templates` folder
-    4. Search on the filesystem for matching directories
-    5. Offer to set up a new one, and make it `init.templateDir`
+   1. From the `$GIT_TEMPLATE_DIR` environment variable
```
Comment (rycus86) [inline README.md]: Same as above.

File: install.sh [hunk 4]
```diff
@@ -2759,7 +2760,11 @@ parse_command_line_arguments() {
             SINGLE_REPO_INSTALL="yes"
         elif [ "$p" = "--skip-install-into-existing" ]; then
             SKIP_INSTALL_INTO_EXISTING="yes"
+        elif [ "$prev_p" = "--template-dir" ] && [ $p != --* ]; then
```
Comment (rycus86) [inline install.sh]: See https://travis-ci.org/rycus86/githooks/jobs/556353067#L525
You could rewrite it like this for example:
```shell
  elif [ "$prev_p" = "--template-dir" ] && (echo "$p" | grep -qE '^\-\-.*'); then
```

File: install.sh [hunk 5]
```diff
@@ -2759,7 +2760,11 @@ parse_command_line_arguments() {
...
+            # Allow user to pass prefered template dir
+            TARGET_TEMPLATE_DIR=$p
```
Comment (rycus86) [inline install.sh]: Can you quote this please to `"$p"`? Thanks

File: install.sh [hunk 6]
```diff
@@ -2759,7 +2760,11 @@ parse_command_line_arguments() {
...
         fi
+        prev_p=$p
```
Comment (rycus86) [inline install.sh]: Same, please quote it.

File: install.sh [hunk 7]
```diff
@@ -2691,6 +2691,7 @@ If you find it useful, please show your support by starring the project in GitHu
 #   0 when successfully finished, 1 if failed
 ############################################################
 execute_installation() {
+    TARGET_TEMPLATE_DIR=""
```
Comment (rycus86) [inline install.sh]: Nit if you don't want to bother, but I'd move this to the start of `parse_command_line_arguments`.

File: install.sh [hunk 8]
```diff
@@ -2931,7 +2936,7 @@ find_git_hook_templates() {
     printf "Do you want to set up a new Git templates folder? [y/N] "
     read -r SETUP_NEW_FOLDER
 
-    if [ "${SETUP_NEW_FOLDER}" = "y" ] || [ "${SETUP_NEW_FOLDER}" = "Y" ]; then
+    if [ "${SETUP_NEW_FOLDER}" = "y" ] || [ "${SETUP_NEW_FOLDER}" = "Y" ] || is_non_interactive; then
```
Comment (rycus86) [inline install.sh]: I'd pull out this
```shell
if is_non_interactive; then
  setup_new_templates_folder
  if [ "$TARGET_TEMPLATE_DIR" != "" ]; then return; fi
fi
```
to before `# 4. try to search for it on disk` so we don't ask questions that won't get answers anyway.

**Comment by AlexGoris-Cashforce** [comment]
@rycus86 mostly good points, thanks! I'll fix them this afternoon.
About the markdown formatting, I'll respond inline on the comments.

**Comment by rycus86** [comment]
Thanks a lot @AlexGoris-Cashforce ! It looks great!
Could you please just squash these into a single commit and force push, then I'll merge it in. :tada: 
(if you like you could also remove the whitespace at https://github.com/rycus86/githooks/pull/20/files#diff-3fbb47e318cd8802bd325e7da9aaabe8R2883 that `shfmt` is screaming about here https://travis-ci.org/rycus86/githooks/jobs/557180717 but I'm happy to fix that one up afterwards)

**Comment by rycus86** [comment]
I merged another PR, so you'll need to rebase or merge and resolve the conflicts before squashing too, sorry.

**Comment by AlexGoris-Cashforce** [comment]
Removed whitespace, rebased on your latest master and squashed all commits. Should be good to go now I think?

**Comment by rycus86** [comment]
Thanks a lot @AlexGoris-Cashforce !
